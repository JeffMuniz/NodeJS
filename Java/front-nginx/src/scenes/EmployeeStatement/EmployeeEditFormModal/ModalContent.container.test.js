import React from "react";
import { create } from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";
import { ModalContent } from "./ModalContent.container";

jest.mock("src/modules/Form/Form", () => "FormWrapper");
jest.mock("./FormContent/FormContent.schema", () => "FormSchema");
jest.mock("./FormContent/FormContent", () => "FormContent");
jest.mock("@common", () => ({
  SvgIcon: "SvgIcon",
}));

describe("Employee Edit Modal - Modal Content Container", () => {
  let testRenderer;
  let testInstance;

  const entity = {
    cpf: "lorem",
    idCompany: 123,
    status: "ativo",
  };

  const registryUpdate = {
    name: "ipsum",
    registry: 321,
  };

  const props = {
    entity,
    updateEmployee: jest.fn(),
    getEmployeeRegistry: jest.fn(),
    closeModal: jest.fn(),
    showToast: jest.fn(),
  };

  beforeEach(() => {
    testRenderer = create(<ModalContent {...props} />);
    testInstance = testRenderer.getInstance();

    testInstance.setState = jest.fn();
  });

  it("Should render ModalContent", () => {
    // given
    const renderer = new ShallowRenderer();

    // when
    renderer.render(<ModalContent {...props} />);
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });

  it('Should update employee and close modal"', async () => {
    // when
    await testInstance.onSubmit(registryUpdate);

    // expect
    expect(testInstance.setState).toBeCalled();
    expect(testInstance.props.updateEmployee).toBeCalledWith(registryUpdate);
    expect(testInstance.props.getEmployeeRegistry).toBeCalled();
    expect(testInstance.props.closeModal).toBeCalled();
  });
});
