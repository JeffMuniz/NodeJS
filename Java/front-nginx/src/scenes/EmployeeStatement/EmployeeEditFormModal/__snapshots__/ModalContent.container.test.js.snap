// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee Edit Modal - Modal Content Container Should render ModalContent 1`] = `
<div>
  <styled.div>
    <styled.span>
      Editar Funcion√°rio
    </styled.span>
  </styled.div>
  <styled.div>
    <FormWrapper
      initialValues={
        Object {
          "cpf": "lorem",
          "idCompany": 123,
          "status": "ativo",
        }
      }
      onSubmit={[Function]}
      render={[Function]}
      validationSchema="FormSchema"
    />
  </styled.div>
</div>
`;
